openssl-dynamic/src/main/java/io/netty/internal/tcnative/NativeStaticallyReferencedJniMethods.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.internal.tcnative;
+
+/**
+ * This class is necessary to break the following cyclic dependency:
+ * <ol>
+ * <li>JNI_OnLoad</li>
+ * <li>JNI Calls FindClass because RegisterNatives (used to register JNI methods) requires a class</li>
+ * <li>FindClass loads the class, but static members variables of that class attempt to call a JNI method which has not
+ * yet been registered.</li>
+ * <li>{@link java.lang.UnsatisfiedLinkError} is thrown because native method has not yet been registered.</li>
+ * </ol>
+ * <strong>Static members which call JNI methods must not be declared in this class!</strong>
+ */
+final class NativeStaticallyReferencedJniMethods {
```

Comment:
this comment was collapsed but no action has yet been taken:

from @nmittler https://github.com/netty/netty-tcnative/pull/239#discussion_r103495559

> Same comment regarding the name ... what about `NativeConstants`?

from me:

> I don't have a strong preference either way but just to explain the rational for the existing name:
> - To clarify that not all constants have to go here (only the statically referenced ones), and that this class exists specifically to avoid class loader issues. So folks should not add things to this class which may introduce a class loader cycle described in the class comments. For example someone may think "I will also add a `static final int MY_VAR = someJNIMethod();` in this class to make the value easier to access from java." which would be wrong.
> - To be consistent with transport-native-epoll. The plan is to move tcnative into the Netty project so consistently will help folks understand similar concepts across module boundaries.
> Also this class is internal so folks will not be exposed to it.